### ISSUE TITLE
Failure when parsing `foo ? ./${bar}`

### ISSUE THREAD

**Issue by azahi** [Synthetica9/nix-linter#67]
Given this valid Nix expression:

```nix
{
  inputs,
  lib,
}:
with lib; let
  mkConfiguration = name: {
    modules ? [],
    configuration ? ./${name},
    this ? my.configurations.${name},
  }:
    nameValuePair name (nixosSystem {
      inherit (this) system;
      modules =
        attrValues inputs.self.nixosModules
        ++ modules
        ++ [(import configuration)];
      specialArgs = {inherit inputs lib this;};
    });
in
  mapAttrs' mkConfiguration {
    melian.modules = with inputs; [
      nixos-hardware.nixosModules.common-pc-laptop-ssd
      nixos-hardware.nixosModules.lenovo-thinkpad-t480
      nixpkgs.nixosModules.notDetected
    ];
};
```

nix-linter 0.2.0.4 fails with:

```
Failure when parsing:
configurations/default.nix:8:23:
  |
8 |     configuration ? ./${name},
  |                       ^
unexpected '$'
expecting '!', '-', or path
```

Not sure if I must report it here or in the hnix repo.
